dockertest/output/dockerinfo.py

```diff
@@ -0,0 +1,104 @@
+"""
+Handler for 'docker info'. Runs command, returns output in
+machine-friendly form.
+"""
+
+import collections
+import subprocess
+
+
+class DockerInfo(collections.MutableMapping):
+    """
+    Parser of 'docker info' output
+    """
+
+    def __init__(self, info_string=None, docker_path=None):
+        if info_string is None:
+            if docker_path is None:
+                docker_path = 'docker'
+            info_string = subprocess.check_output(docker_path + ' info',
+                                                  shell=True,
+                                                  close_fds=True)
+        self.info_string = info_string
+        self.table = self._build_table(info_string)
+
+    @staticmethod
+    def _build_table(info_string):
+        """
+        'docker info' returns a human-readable list of key: value pairs.
+        Some are indented by a space, indicating that these key/values
+        are subelements of a previous element. To wit:
+
+             Images: 3
+             Server Version: 1.12.6
+             Storage Driver: devicemapper
+              Pool Name: vg--docker-docker--pool
+              Pool Blocksize: 524.3 kB
+              ...
+             Logging Driver: journald
+
+        In this case 'Images', 'Server Version', and 'Logging Driver'
+        are simple tuples but 'Storage Driver' has both a value and
+        a set of further key/value pairs: Pool Name, Pool Blocksize, ...
+
+        We parse that and create a dict with the expected key/value
+        mapping *and* an extra: for all elements with subelements,
+        'element...' (element name plus three dots) is a dict containing
+        the subelements. E.g. x['Storage Driver...']['Pool Name'] = 'vg--etc'
+        """
+        table = {}
+        current_key = None
+        for line in info_string.splitlines():
+            # Almost every line will be Foo: Bar, but 'Insecure Registries:'
+            # is followed by a simple list of IPv4 netmasks
+            if ': ' in line or line.endswith(':'):
+                (key, value) = [e.strip() for e in line.split(':', 1)]
+            else:
+                key = line.strip()
+                value = ''
+            if line.startswith(' '):
+                if not current_key:
+                    raise IndexError("Internal error: indented output line"
```

Comment:
Fixed. Removed "Internal error" from the string. Changed to `SyntaxError` instead; it seemed most appropriate than RuntimeError.